illumos-utils/src/opte/illumos/port_manager.rs

```diff
@@ -490,6 +493,53 @@ impl PortManager {
         }
         Ok(())
     }
+
+    pub fn set_virtual_nic_host(
+        &self,
+        _interface_id: Uuid,
+        mapping: &SetVirtualNetworkInterfaceHost,
+    ) -> Result<(), Error> {
+        let hdl = OpteHdl::open(OpteHdl::XDE_CTL)?;
+
+        hdl.set_v2p(&SetVirt2PhysReq {
+            vip: match mapping.virtual_ip {
+                std::net::IpAddr::V4(ipv4) => oxide_vpc::api::IpAddr::Ip4(
+                    oxide_vpc::api::Ipv4Addr::from(ipv4),
+                ),
+                std::net::IpAddr::V6(ipv6) => oxide_vpc::api::IpAddr::Ip6(
+                    oxide_vpc::api::Ipv6Addr::from(ipv6),
+                ),
+            },
+            phys: PhysNet {
+                ether: oxide_vpc::api::MacAddr::from(
+                    (*mapping.virtual_mac).into_array(),
+                ),
+                ip: match mapping.physical_host_ip {
+                    std::net::IpAddr::V4(_) => {
+                        return Err(opte_ioctl::Error::InvalidArgument(
+                            String::from("sleds should only have v6 addresses"),
+                        )
+                        .into());
+                    }
```

Comment:
Any reason not to just leave `physical_host_ip` as an `Ipv6Addr`? That's what we're getting when you look up the sled info to make this request to begin with.

Reply:
Nope, other than `IpAddr` matches what's in OPTE:

```rust
pub struct SetVirt2PhysReq {
    pub vip: IpAddr,
    pub phys: PhysNet,
}
```

Reply:
Wait! Wrong struct.

Reply:
changed to Ipv6Addr in f015b252